diff --git a/backend/dvadmin/system/views/message_center.py b/backend/dvadmin/system/views/message_center.py
index 67e547b3..0848bea6 100644
--- a/backend/dvadmin/system/views/message_center.py
+++ b/backend/dvadmin/system/views/message_center.py
@@ -116,21 +116,27 @@ class MessageCenterCreateSerializer(CustomModelSerializer):
             users = Users.objects.filter(dept__id__in=target_dept).values_list('id', flat=True)
         if target_type in [3]:  # 系统通知
             users = Users.objects.values_list('id', flat=True)
-            websocket_push("dvadmin", message={"sender": 'system', "contentType": 'SYSTEM',
-                                               "content": '您有一条新消息~', "refresh_unread": True})
+            
         targetuser_data = []
         for user in users:
             targetuser_data.append({
                 "messagecenter": data.id,
                 "users": user
             })
-            if target_type in [1,2]:
-                room_name = f"user_{user}"
-                websocket_push(room_name, message={"sender": 'system', "contentType": 'SYSTEM',
-                                                   "content": '您有一条新消息~', "refresh_unread": True})
         targetuser_instance = MessageCenterTargetUserSerializer(data=targetuser_data, many=True, request=self.request)
         targetuser_instance.is_valid(raise_exception=True)
         targetuser_instance.save()
+        
+        # 保存后再推送消息，避免前端获取未读数时数据还未保存
+        if target_type in [3]:
+            websocket_push("dvadmin", message={"sender": 'system', "contentType": 'SYSTEM',
+                                               "content": '您有一条新消息~', "refresh_unread": True})
+        else:
+            for user in users:
+                if target_type in [0, 1, 2]:
+                    room_name = f"user_{user}"
+                    websocket_push(room_name, message={"sender": 'system', "contentType": 'SYSTEM',
+                                                       "content": '您有一条新消息~', "refresh_unread": True})
         return data
 
     class Meta:
@@ -173,7 +179,7 @@ class MessageCenterViewSet(CustomModelViewSet):
         # 主动推送消息
         room_name = f"user_{user_id}"
         websocket_push(room_name, message={"sender": 'system', "contentType": 'TEXT',
-                                 "content": '您查看了一条消息~', "refresh_unread": True})
+                                 "content": '', "refresh_unread": True})
         return DetailResponse(data=serializer.data, msg="获取成功")
 
     @action(methods=['GET'], detail=False, permission_classes=[IsAuthenticated])
diff --git a/web/src/api/websocket.js b/web/src/api/websocket.js
index 4dbc3a39..9c908a1e 100644
--- a/web/src/api/websocket.js
+++ b/web/src/api/websocket.js
@@ -29,15 +29,22 @@ function webSocketOnError (e) {
  */
 function webSocketOnMessage (e) {
   const data = JSON.parse(e.data)
-  const { refreshUnread, systemConfig } = data
+  const refreshUnread = data.refreshUnread || data.refresh_unread
+  const systemConfig = data.systemConfig || data.system_config
   if (refreshUnread) {
     // 更新消息通知条数
     store.dispatch('d2admin/messagecenter/setUnread')
   }
   if (systemConfig) {
     // 更新系统配置
-    this.$store.dispatch('d2admin/settings/load')
+    store.dispatch('d2admin/settings/load')
   }
+  
+  // 如果没有内容则不弹窗
+  if (!data.content) {
+    return
+  }
+
   if (data.contentType === 'SYSTEM') {
     ElementUI.Notification({
       title: '系统消息',
